packages/fluentui/react-northstar/src/components/Chat/ChatMessage.tsx

```diff
@@ -477,23 +484,62 @@ export const ChatMessage: ComponentWithAs<'div', ChatMessageProps> &
   });
 
   const detailsElement = createShorthand(ChatMessageDetails, details, {
-    defaultProps: () => ({ mine }),
+    defaultProps: () => ({ compact, mine }),
   });
 
-  const readStatusElement = createShorthand(ChatMessageReadStatus, readStatus, {});
+  const readStatusElement = createShorthand(ChatMessageReadStatus, readStatus, {
+    defaultProps: () => ({ compact }),
+  });
 
-  const headerElement = createShorthand(ChatMessageHeader, header || {}, {
-    overrideProps: () => ({
-      content: (
-        <>
-          {authorElement}
+  let layout = <></>;
+  if (compact) {
+    const headerElement = createShorthand(ChatMessageHeader, header);
+
+    layout = (
+      <>
+        {actionMenuElement}
+        <div className={chatMessageSlotClassNames.bar} />
```

Comment:
Hi, I have a concern on the new slots - they are not extensible. 
Normally fluent slots has their own slot component, or utilizes the `create` function of other fluent component, which enables slots to take any custom prop passed in by user. 
For example, timestamp in chat messages uses `Text.create` to create its own slot component.
If I want to add an `aria-hidden` to the `timestamp` slot, it can be done so by passing `timestamp={{ content: 'Yesterday, 00:15 PM', 'aria-hidden': true }}` prop to `ChatMessage` component, and I would be able to see `aria-hidden` in the final DOM.
However the new slots added this way here could not be customized like the existing slots in ChatMessages.

We discussed within the team, and if you're absolutely sure there will be no custom prop/a11y behavior needed for these slots, then we're okay with the current state. If not, the better way is to make the slots slot components. 
You could follow the timestamp example, which uses `create` function.
And here's an example which adds dedicated slot component: https://github.com/microsoft/fluentui/pull/13776/files adds details slot and ChatMessageDetails slot component.

Reply:
Added a proper slot for `compactBody` and removed the `main` slot.